test/extended/images/helper.go

```diff
@@ -16,3 +24,48 @@ func GetImageLabels(c client.ImageStreamImageInterface, imageRepoName, imageRef
 	}
 	return image.Image.DockerImageMetadata.Config.Labels, nil
 }
+
+// ModifySourceCode will modify source code in the pod of the application
+// according to the sed script.
+func ModifySourceCode(oc *exutil.CLI, selector labels.Selector, sedScript, file string) error {
+	pods, err := exutil.WaitForPods(oc.KubeREST().Pods(oc.Namespace()), selector, exutil.CheckPodIsRunningFunc, 1, 120*time.Second)
+	if err != nil {
+		return err
+	}
+	if len(pods) != 1 {
+		return fmt.Errorf("Got %d pods for selector %v, expected 1", len(pods), selector)
+	}
+
+	pod, err := oc.KubeREST().Pods(oc.Namespace()).Get(pods[0])
+	if err != nil {
+		return err
+	}
+	return oc.Run("exec").Args(pod.Name, "-c", pod.Spec.Containers[0].Name, "--", "sed", "-ie", sedScript, file).Execute()
+}
+
+// CheckPageContains makes a http request for an example application and checks
+// that the result contains given string
+func CheckPageContains(oc *exutil.CLI, endpoint, contents string) (bool, error) {
+	address, err := exutil.GetEndpointAddress(oc, endpoint)
+	if err != nil {
+		return false, err
+	}
+
+	// Make the request
+	resp, err := http.Get(fmt.Sprintf("http://%s/", address))
```

Comment:
@mfojtik yes. CheckPageContains() calls GetEndpointAddress() which calls WaitForAnEndpoint(). That is equivalent to the code @bparees linked above.